core/src/test/java/com/datastax/oss/driver/internal/core/context/StartupOptionsBuilderTest.java

```diff
@@ -87,22 +61,43 @@ private void assertDefaultStartupOptions(Startup startup) {
   }
 
   @Test
-  public void should_build_minimal_startup_options() {
-    buildDriverContext();
-    Startup startup = new Startup(defaultDriverContext.getStartupOptions());
+  public void should_build_startup_options_with_no_compression_if_undefined() {
+
+    DefaultDriverContext ctx = MockedDriverContextFactory.defaultDriverContext();
+    Startup startup = new Startup(ctx.getStartupOptions());
     assertThat(startup.options).doesNotContainKey(Startup.COMPRESSION_KEY);
     assertDefaultStartupOptions(startup);
   }
 
   @Test
-  public void should_build_startup_options_with_compression() {
-    when(defaultProfile.isDefined(DefaultDriverOption.PROTOCOL_COMPRESSION))
-        .thenReturn(Boolean.TRUE);
-    when(defaultProfile.getString(DefaultDriverOption.PROTOCOL_COMPRESSION)).thenReturn("lz4");
-    buildDriverContext();
-    Startup startup = new Startup(defaultDriverContext.getStartupOptions());
-    // assert the compression option is present
-    assertThat(startup.options).containsEntry(Startup.COMPRESSION_KEY, "lz4");
+  public void should_build_startup_options_with_no_compression_if_defined_as_none() {
+
+    DefaultDriverContext ctx = buildMockedContext("none");
+    Startup startup = new Startup(ctx.getStartupOptions());
+    assertThat(startup.options).doesNotContainKey(Startup.COMPRESSION_KEY);
     assertDefaultStartupOptions(startup);
   }
+
+  @Test
+  public void should_build_startup_options_with_lz4_compression() {
+
+    doCompressorStartupOptionsTest("lz4");
+  }
+
+  @Test
+  public void should_build_startup_options_with_snappy_compression() {
+
+    doCompressorStartupOptionsTest("snappy");
```

Comment:
I think you can replace those one line tests with one parameterized test

Reply:
You're thinking of... the built-in JUnit support for parameterized tests @tomekl007 ?  Yeah, I think that could be made to work... but there are some things I'm not super fond of there:

* The reliance on test state (via either the constructor usage or injection) isn't super-awesome
* The inability to mix and match parameterized vs. other tests in the same class seems problematic

Seems like the only way to address that last problem would be to create a distinct test just for these two cases... and that seems like an increase in complexity.

Or were you thinking of something more along the lines of https://github.com/Pragmatists/JUnitParams?  That addresses both concerns above with a much simpler API.  I could get behind that... is it a big issue to add that as another dep?

Reply:
We are already using parameterized tests in this project via: 
`com.tngtech.java.junit.dataprovider`
(see for example `DataTypeIT`) - I think we should stick to that library and approach of parameterized tests :)

Reply:
Nice!  I think that takes care of my concerns mentioned above... will gladly switch to that.  Thanks for the pointer @tomekl007 !